### ISSUE TITLE
Future of Compare plugin

### ISSUE THREAD

**Issue by Ede123** [jsleroy/compare-plugin#2]
Currently there seem to be at least two branches of the Compare plugin
* The version found here and at https://sourceforge.net/projects/npp-compare/ respectively by @jsleroy.
* Some versions at https://sourceforge.net/projects/sourcecookifier/files/other%20plugins/ by @ufo.
* Some other versions at https://bitbucket.org/uph0/compare/ (also by @ufo).

There are multiple issues though:
* Both branches seem to have some (but likely not all) updates.
* Conflicting version numbers (1.5.6.2 is the latest available from SourceForge by @jsleroy, 1.5.6.4 is the latest available from SourceForge by @ufo, 1.5.6.6 is available from Bitbucket by @ufo, this repository lists 1.5.7 which is not available anywhere in binary form though).
* Notepad++ PluginManager does not now about any of those updated versions and provides users with an outdated version 1.5.6 that has some issues.

I think it would be great if both branches could be merged, including
* One common code repository (e.g. this one or the one on Bitbucket by @ufo)
* One "official" homepage (e.g. the current SourceForge project page at https://sourceforge.net/projects/npp-compare/ or simply the GitHub or the Bitbucket project page if you want to drop SourceForge as Notepad++ and many other projects did in the meantime)
* Links that point to the "official" homepage of choice from inside the plugin and from all other project pages that might have existed at some time, so it is clear where the most recent information and the most recent code/binaries can be found.
* Once the above is done one should also announce this on https://notepad-plus-plus.org/community/category/5/plugin-development and update the information for PluginManager, so people will get the latest version.

In this way I think popularity of the Compare plugin will greatly increase (right now it might seem to users that the plugin is unmaintained since the Bitbucket repository is not among the top hits of any search engine and the other repositories were not updated in a long time).
Also contributions can be channeled much more efficiently greatly benefiting the project.

Please let me know what you think of this suggestion. If you're willing to follow my suggestions I'd be happy to support you if you can use help with any of the steps.

**Comment by jsleroy** [comment]
Hi @Ede123,

First, thank you for this much appreciated initiative. To be honest, I don't have that much time for the moment to provide decent support/update of the Compare plugin. So, trying to find some alternatives to restore the level of update/support as it was before is appreciated.

Actually, I was not aware of the forked repository that @ufo created. The bitbucket repository seems to contains the most up-to-date version of the source files, so I've updated the github repo to includes @ufo's changes.

* I would like to keep the github repository as the official one, as it used to be since a decent time now.
* The sourceforge repository is not maintained anymore so I will consider shutting it down.
I should also move the Sourceforge webpage on github, probably using a wiki page.
* Links in the source code should be updated to point to the github repository.
* Old released plugins binary hosted on Sourceforge should be moved on github as well, however I'm not sure how to do that (I've never used github for binary hosting).

I would like to keep the ownership of the plugin, hence my preference of keeping the official repository on github. But I would greatly appreciated to add @ufo as a github collaborator, so that he can freely update the plugin, make releases and provide support.

As @ufo seem to be mostly inactive on github, would you mind sending him an email ?

@Ede123, Would you like/be able to help on some of these topics ?

Best regards

**Comment by Ede123** [comment]
Great to hear you're up for it!

I had already opened [an issue](https://bitbucket.org/uph0/compare/issues/6/) on ufo's Bitbucket repository to catch his attention. Since I'm not able to find an e-mail address anywhere I don't know of a better way to contact him.

I hope he's on board, too, and of course I'll be happy to help out myself where needed (anything specific you'd like my support on?).

Regarding binary hosting: Releases on GitHub are bound to tags. For every tag one can create a release and attach files to it.

**Comment by jsleroy** [comment]
Let's wait for @ufo to reply.

I would be more than happy if you could help on the following points:

* Migrating Sourceforge issues/feature requests on Github (I've had a quick look at what exists and it seems there is some automated way to do that but don't have much time to investigate that).
* Migrating releases to github/adding support in github to handle the releases.
* Migrating the "kind of" webpage I had on Sourceforge (or better, designing a new one, I've always dreamed about a real webpage that looks nice but don't have the knowledge to do it).
* Fixing bugs/adding features.
* Contacting the Plugin Manager owner to get Compare back to its list of supported plugins.
* Communicating on Compare/adding more visibility (I have a facebook page actually that I can share with you).
* Helping on any other points I'm not thinking about.
